src/tags/tests/test_models.py

```diff
@@ -203,6 +260,27 @@ def test_query_tag_from_approved_quest(self):
         self.assertTrue('approved' in tags)
         self.assertFalse('submitted' in tags)
 
+    def test_total_xp_is_correct(self):
+        """ check if get_user_tags_and_xp's xp adds up correctly """
+
+        # quest only
+        quest1, _ = self.create_quest_and_submissions(2)
+        quest1.tags.add('tag-1')
+
+        # badge only
+        badge1, _ = self.create_badge_and_assertions(3)
+        badge1.tags.add('tag-2')
+
+        # quest + badge
+        quest2, _ = self.create_quest_and_submissions(5)
+        badge2, _ = self.create_badge_and_assertions(7)
+        quest2.tags.add('tag-3')
+        badge2.tags.add('tag-3')
+
+        total_xp = sum(t[1] for t in get_user_tags_and_xp(self.user))
```

Comment:
Everything looks good, but going to take this opportunity for you to learn about Django queryset aggregate functions.  Whenever we can, we want to avoid list comprehension or looping through lists/querysets.  Django has optimized functions that can do this in a single step.  Even though this is a tiny list and doesn't really matter, it's a good opportunity to practice.  Take a look at the docs here, and see if you can figure out how to do it (if you can, maybe you can't in this case, not sure what get_user_tags_and_xp returns) https://docs.djangoproject.com/en/5.0/ref/models/querysets/#aggregate

@coderabbitai can you provide a suggestion to optimize this using Django's aggregate function?  You should go into the code and look at `get_user_tags_and_xp` to see what kind of data it returns.

Reply:
```get_user_tags_and_xp``` returns a list of tuples. each tuple holding tag and total xp for that tag
![image](https://github.com/bytedeck/bytedeck/assets/39788517/06a45ffa-cab4-4eb1-82b4-694df7945121)

Reply:
i dont think aggregate will work here